ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -304,6 +306,44 @@ def _llmobs_set_meta_tags_from_similarity_search(
                 except TypeError:
                     log.warning("Failed to serialize similarity output documents to JSON")
 
+    def _llmobs_set_meta_tags_from_tool(
+        self,
+        span: Span,
+        tool_inputs: Union[str, Dict[str, object], object],
+        tool_output: object,
+        error: bool,
+    ) -> None:
+        metadata = span.get_tag(METADATA) or {}
+
+        span.set_tag_str(SPAN_KIND, "tool")
+        if tool_inputs is not None:
+            tool_input = tool_inputs.get("input")
+            if tool_inputs.get("config"):
+                metadata["tool_config"] = tool_inputs.get("config")
+            if tool_inputs.get("info"):
+                metadata["tool_info"] = tool_inputs.get("info")
+            if metadata:
+                span.set_tag_str(METADATA, json.dumps(metadata))
```

Comment:
Can we be safe with json dumping unknown config dicts (which might contain non-json serializable objects?)
```python
from ddtrace.llmobs._utils import _unserializable_default_repr

span.set_tag_str(METADATA, json.dumps(metadata, default=_unserializable_default_repr))
```